contracts/root/depositManager/DepositManager.sol

```diff
@@ -56,4 +59,36 @@ contract DepositManager is DepositManagerStorage, IDepositManager {
   {
     rootChain.createDepositBlock(_user, _token, amountOrNFTId);
   }
+
+  function tokenFallback(address _user, uint256 _amount, bytes _data) public {
+    address _token = msg.sender;
+
+    // create deposit block with token fallback
+    depositManager.createDepositBlock(_currentHeaderBlock, _token, _user, _amount);
+  }
+
+  function onERC721Received(address operator, address from, uint256 tokenId, bytes data) public returns (bytes4) {
+    depositManager.createDepositBlock(_currentHeaderBlock, msg.sender, from, tokenId);
+    return 0x150b7a02;
+  }
+    // transfer tokens to user
+  function transferAmount(
+    address _token,
+    address _user,
+    uint256 _amount
+  ) public onlyWithdrawManager returns(bool)  {
+
+    address wethToken = depositManager.wethToken();
+
+    // transfer to user TODO: use pull for transfer
+    if (depositManager.isERC721(_token)) {
```

Comment:
bhai ek baar PR compile toh krke dekh lo

Reply:
we gotta first conclude on where to put that mapToken thing.